Source/TrackableEntities.EF.5/DbContextExtensions.cs

```diff
@@ -199,16 +199,21 @@ public static void ApplyChanges(this InterceptorPool pool, IEnumerable<ITrackabl
                     context.ApplyChangesOnProperties(item, visitationHelper.Clone(), TrackingState.Deleted, interceptors);
                 }
 
+                var isComplex = context.IsComplexType(item.GetType());                
+                
                 // Set modified properties
-                if (item.TrackingState == TrackingState.Modified
-                    && (state == null || state == TrackingState.Modified)
-                    && item.ModifiedProperties != null
-                    && item.ModifiedProperties.Count > 0)
+                if (isModifiable(isComplex))
                 {
                     // Mark modified properties
                     SetEntityState(context, item, parent, propertyName, EntityState.Unchanged, interceptors);
-                    foreach (var property in item.ModifiedProperties)
-                        context.Entry(item).Property(property).IsModified = true;
+                    var entry = isComplex ? context.Entry(parent) : context.Entry(item);
+                    if(isComplex && entry.State == EntityState.Detached) entry.State = parent.TrackingState.ToEntityState();
```

Comment:
OK, so I commented out line [210](https://github.com/TrackableEntities/trackable-entities/commit/f5df58c264e626d414cc3c968af429d8b7900f61#diff-18c35e3f0eae888189d043a03dbb3ef2R210), and all the unit tests still pass.  If you think this line is necessary, can you create a failing test that will pass when the line is inserted?